cli/cmd/resources/odiglet.go

```diff
@@ -505,6 +510,84 @@ func NewOdigletDaemonSet(ns string, version string, imagePrefix string, imageNam
 			},
 		},
 	}
+
+	// Check for a custom offsets file, if present use it
+	// The reason for this is because a user can update their offsets asynchronously with the Odigos release
+	// For example:
+	// 1) User updates offsets to support Go 1.24
+	// 2) Odigos releases support for Go 1.24
+	// 3) User updates offsets to support Go 1.24.1
+	// 4) User updates Odigos (no default support for 1.24.1)
+	// This also means a user could be still using old offsets, and might want to revert that to pull in latest Odigos changes
+	// TODO: Detect superceded offsets automatically so user doesn't need to manually revert
+	_, err := client.Clientset.CoreV1().ConfigMaps(ns).Get(ctx, consts.GoOffsetsConfigMap, metav1.GetOptions{})
+	if err == nil {
+		// Add offsets volume (if not already exist)
+		volumes := ds.Spec.Template.Spec.Volumes
+		if volumes == nil {
+			volumes = make([]v1.Volume, 0)
+		}
+		addVolume := true
+		for _, vol := range volumes {
+			if vol.Name == consts.GoOffsetsConfigMap {
+				addVolume = false
+				break
+			}
+		}
+		if addVolume {
+			volumes = append(volumes, v1.Volume{Name: consts.GoOffsetsConfigMap,
+				VolumeSource: v1.VolumeSource{
+					ConfigMap: &v1.ConfigMapVolumeSource{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: consts.GoOffsetsConfigMap,
+						},
+					},
+				},
+			})
+			ds.Spec.Template.Spec.Volumes = volumes
+		}
+
+		// Add offsets volume mounnt (if not already exist)
+		volumeMounts := ds.Spec.Template.Spec.Containers[0].VolumeMounts
+		if volumeMounts == nil {
+			volumeMounts = make([]v1.VolumeMount, 0)
+		}
+		addVolumeMount := true
+		for _, vm := range volumeMounts {
+			if vm.Name == consts.GoOffsetsConfigMap {
+				addVolumeMount = false
+				break
+			}
+		}
+		if addVolumeMount {
+			volumeMounts = append(volumeMounts, v1.VolumeMount{Name: consts.GoOffsetsConfigMap, MountPath: offsetFileMountPath})
+			ds.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
+		}
+
+		// Add offsets Env Var (if not already exist)
+		envVars := ds.Spec.Template.Spec.Containers[0].Env
+		if envVars == nil {
+			envVars = make([]v1.EnvVar, 0)
+		}
+		addEnvVar := true
+		for _, env := range envVars {
+			if env.Name == consts.GoOffsetsEnvVar {
+				addEnvVar = false
+				break
+			}
+		}
+		if addEnvVar {
+			envVars = append(envVars, v1.EnvVar{Name: consts.GoOffsetsEnvVar, Value: offsetFileMountPath + "/" + consts.GoOffsetsFileName})
+			ds.Spec.Template.Spec.Containers[0].Env = envVars
+		}
+	} else {
+		if !apierrors.IsNotFound(err) {
+			fmt.Println(fmt.Sprintf("\033[31mERROR\033[0m Unable to get Go offsets ConfigMap: %s", err))
+			os.Exit(1)
```

Comment:
as mentioned in another comment, I think we should only `os.Exit` at the top, so it will not surprise anyone in the future. especially since these are used in install, and upgrade which are more sensitive commands